src/_pytest/mark/structures.py

```diff
@@ -153,8 +161,19 @@ def combined_with(self, other):
         combines by appending args and merging the mappings
         """
         assert self.name == other.name
+
```

Comment:
we might want to add a comment that this is intended to be a temporary hack to be resolved with object based marks

Reply:
:+1: 
Will revisit before merging.

Thanks for the approval - I'm glad we could figure out something in the end.